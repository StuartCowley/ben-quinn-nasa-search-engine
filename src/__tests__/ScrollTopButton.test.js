import React from "react";
import { render } from "@testing-library/react";
import ScrollTopButton from "../components/SrollTopButton";

describe("ScrollTopButton", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<ScrollTopButton />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders when user scrolls past veiwport height", async () => {
    const { getByTestId } = render(<ScrollTopButton isVisible />);

    expect(getByTestId("scroll-top-button")).toBeVisible();
  });
});
