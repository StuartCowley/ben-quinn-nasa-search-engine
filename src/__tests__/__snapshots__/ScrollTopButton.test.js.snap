// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollTopButton renders correctly 1`] = `
<DocumentFragment>
  <button
    class="scroll-top-button"
    data-testid="scroll-top-button"
    style="display: none;"
    type="submit"
  >
    Scroll to top
  </button>
</DocumentFragment>
`;
